buildcontext/gitlookup.go

```diff
@@ -155,6 +188,47 @@ func (gl *GitLookup) getGitMatcher(path string) (string, *gitMatcher, error) {
 	return "", nil, ErrNoMatch
 }
 
+// detectProtocol will update the gitMatcher protocol if it is set to auto
+func (gl *GitLookup) detectProtocol(host string) (protocol string, err error) {
```

Comment:
This might be a nit; but the only thing I would do in addition here is maybe enumerate the protocols to make it clearer down the line. Like:

```
type gitProtocol string

const (
	auto gitProtocol = "auto"
	ssh gitProtocol = "ssh"
)
```

This would ensure that git protocols stay on the rails? At the very least I would move them into a constants list and name them, the type just helps them only get used where they should be used.

Reply:
that's a fair point, i'll make a follow up PR for that. (I have some other git-related changes I'm currently working on which would cause a bit more rebasing issues).